libsolidity/ast/Types.cpp

```diff
@@ -2353,11 +2304,11 @@ u256 StructType::memoryDataSize() const
 	return size;
 }
 
-bigint StructType::storageSizeUpperBound() const
+bigint StructType::sizeUpperBound(DataLocation _loc) const
 {
-	bigint size = 1;
-	for (auto const& member: members(nullptr))
-		size += member.type->storageSizeUpperBound();
+	bigint size = (_loc == DataLocation::Storage) ? 1 : 0;
```

Comment:
Just to make people extra aware of what's happening here, I'd add a comment that its 1 **slot** vs 0 **bytes**.

Reply:
Done.